examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
+    MICROSTRAIN_LOG_INFO("Setting the device to idle.\n");
+    cmd_result = mip_base_set_idle(_device);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not set the device to idle!\n");
+    }
+
+    // Print device info to make sure the correct device is being used
+    MICROSTRAIN_LOG_INFO("Getting the device information.\n");
+    mip_base_device_info device_info;
+    cmd_result = mip_base_get_device_info(_device, &device_info);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not get the device information!\n");
+    }
+
+    // Extract the major minor and patch values
+    const uint16_t major = device_info.firmware_version / 1000;
+    const uint16_t minor = device_info.firmware_version / 100 % 10;
+    const uint16_t patch = device_info.firmware_version % 100;
+
+    // Firmware version format is x.x.xx
+    char firmwareVersion[16];
+    snprintf(firmwareVersion, sizeof(firmwareVersion) / sizeof(firmwareVersion[0]), "%d.%d.%02d",
+        major,
+        minor,
+        patch
+    );
+
+    MICROSTRAIN_LOG_INFO("-------- Device Information --------\n");
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Name", device_info.model_name);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Model Number", device_info.model_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Serial Number", device_info.serial_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Lot Number", device_info.lot_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Options", device_info.device_options);
+    MICROSTRAIN_LOG_INFO("%-16s | %16s\n", "Firmware Version", firmwareVersion);
+    MICROSTRAIN_LOG_INFO("------------------------------------\n");
+
+    // Load the default settings on the device
+    // Note: This guarantees the device is in a known state
+    MICROSTRAIN_LOG_INFO("Loading default settings.\n");
+    cmd_result = mip_3dm_default_device_settings(_device);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not load device default settings!\n");
+    }
+}
+
 ////////////////////////////////////////////////////////////////////////////////
+/// @brief Handles graceful program termination and cleanup
+///
+/// @details Handles graceful shutdown when errors occur:
+///          - Outputs provided error message
+///          - Closes device connection if open
+///          - Exits with appropriate status code
+///
+/// @param _device_port Serial port connection to close
+/// @param _message Error message to display
+/// @param _successful Whether termination is due to success or failure
+///
+void terminate(serial_port* _device_port, const char* _message, const bool _successful)
+{
+    if (strlen(_message) != 0)
+    {
+        if (_successful)
+        {
+            MICROSTRAIN_LOG_INFO("%s", _message);
```

Comment:
Could use `MICROSTRAIN_LOG_LOG(_successful ? MICROSTRAIN_LOG_LEVEL_INFO : MICROSTRAIN_LOG_LEVEL_ERROR, "%s", _message);` instead of the if/else. No preference personally.